src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx

```diff
@@ -0,0 +1,165 @@
+import React from "react";
+import styled from "@emotion/styled/macro";
+import { Disclaimer } from "@leafygreen-ui/typography";
+import Button, { Variant } from "@leafygreen-ui/button";
+import { ProjectVariants } from "gql/queries/patch";
+import { VariantTasksState } from "pages/configurePatch/ConfigurePatchCore";
+import Checkbox from "@leafygreen-ui/checkbox";
+import { css } from "@emotion/core";
+import isEmpty from "lodash/isEmpty";
+
+interface Props {
+  variants: ProjectVariants;
+  selectedBuildVariant: string;
+  selectedVariantTasks: VariantTasksState;
+  setSelectedVariantTasks: React.Dispatch<
+    React.SetStateAction<VariantTasksState>
+  >;
+}
+interface TasksState {
+  [task: string]: true;
+}
+
+export const ConfigureTasks: React.FC<Props> = ({
+  variants,
+  selectedBuildVariant,
+  selectedVariantTasks,
+  setSelectedVariantTasks,
+}) => {
+  const buildVariantCount = Object.keys(selectedVariantTasks).length;
+  const taskCount = Object.values(selectedVariantTasks).reduce(
+    (prev, curr) => prev + Object.values(curr).length,
+    0
+  );
+
+  const projectVariantTasksMap: {
+    [variant: string]: string[];
+  } = variants.reduce((prev, { name, tasks }) => {
+    prev[name] = tasks;
+    return prev;
+  }, {});
+  const currentTasks =
+    projectVariantTasksMap[selectedBuildVariant || variants[0].name];
+
+  const onClickSelectAll = () => {
+    const allTasksForVariant: TasksState = currentTasks.reduce((prev, curr) => {
+      prev[curr] = true;
+      return prev;
+    }, {});
+    setSelectedVariantTasks({
+      ...selectedVariantTasks,
+      [selectedBuildVariant]: allTasksForVariant,
+    });
+  };
+  const onClickDeselectAll = () => {
+    const nextSelectedVariantTasks = { ...selectedVariantTasks };
+    delete nextSelectedVariantTasks[selectedBuildVariant];
+    setSelectedVariantTasks(nextSelectedVariantTasks);
+  };
+
+  const getTaskCheckboxChangeHandler = (task: string, variant: string) => (
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const { checked } = e.target;
+    const nextVariantTasks = { ...selectedVariantTasks[variant] };
+    if (checked) {
+      setSelectedVariantTasks({
+        ...selectedVariantTasks,
+        [variant]: { ...nextVariantTasks, [task]: true },
+      });
+    } else {
+      delete nextVariantTasks[task];
+      const nextSelectedVariantTasks = {
+        ...selectedVariantTasks,
+        [variant]: nextVariantTasks,
+      };
+      if (isEmpty(nextVariantTasks)) {
+        delete nextSelectedVariantTasks[variant];
+      }
+      setSelectedVariantTasks(nextSelectedVariantTasks);
+    }
+  };
+
+  console.log("selectedVariantTasks :", selectedVariantTasks);
+
+  return (
+    <TabContentWrapper>
+      <Actions>
+        <Button variant={Variant.Primary}>Schedule</Button>
+        <ButtonLink
+          data-cy="configurePatch-selectAll"
+          onClick={onClickSelectAll}
+        >
+          Select All
```

Comment:
Ben recommended that these should be checkboxes with an indeterminate state

Reply:
the original design was incorrect and so john and i decided on another design. ben came up with the partial checkbox after the feature had already been implemented. i told him i can make a ticket to do the partial checkbox eventually.

Reply:
cool I made the ticket here: https://jira.mongodb.org/browse/EVG-7871

Reply:
bless